test-app/lib/ember-blog/addon/controllers/post.js

```diff
@@ -35,4 +35,22 @@ export default class extends Controller {
       set(this, 'transitionTo', true);
     });
   }
+
+  @action goToChineseVersion() {
+    this.transitionTo({ queryParams: { lang: 'Chinese' } });
+  }
+
+  @action transitionToHome() {
+    this.transitionToExternal('home').then(() => {
+      var postController = this.controllerFor(this.routeName);
+      postController.set('transitionedToExternal', true);
+    });
+  }
+
+  @action replaceWithHome() {
+    this.replaceWithExternal('home').then(() => {
+      var postController = this.controllerFor(this.routeName);
+      postController.set('replacedWithExternal', true);
+    });
+  }
```

Comment:
Why are we moving it from route to controller?

Reply:
route actions are deprecated